### DISCUSSION TITLE
[Idea] Implement a pluign that supports static lnurl-pay QR so that merchants can accept payments in their store offline

### DISCUSSION THREAD

**Discussion by pavlenex** [btcpayserver/btcpayserver#2617] [Plugins & integrations]
So far, BTCPay has targeted online merchants, our way to retail was always Point Of Sale, however, it's not perfect for every use-case. A barbershop, street artists, aren't really being 

A plugin idea, an "app" where people can create items/prices (similar to PoS) which would then export LNURL QR Codes as an output that merchants can print

They can then just add these QR codes on items in their store

https://youtu.be/_XAvM_LNsoo

https://github.com/lnbits/lnbits/blob/master/lnbits/extensions/offlineshop/README.md

**Comment by ghost** [comment]
but the lightning node still has to be online, no? 
Is there a scenario where lightning node is online, but BTCPay is not?

**Reply by pavlenex** [reply]
An individual can host an instance for the local community members who aren't tech-savvy. I haven't looked into the specs on how it works beyond the links I provided. The idea here isn't to have btcpay offline, it's to avoid having an electronic device on spot to accept payments (static QR code using LNURL)

**Reply by Zaxounette** [reply]
So where a traditionnal BTCPay merchant setup has one BTCPay per merchant _or merchant community_ with each merchant having an internet capable device (tablet, smartphone, etc...) to load the POS app, this idea deletes the second part completely?

Non-internet connected merchants, here we come!

**Reply by pavlenex** [reply]
I think so. Doesn't eliminate the fact they need node/server, so we'll have to figure out node-sharing. @fiatjaf should be able to tell you more, and I think @Kukks can have a perspective on how to achieve this plugin-wise.

**Reply by fiatjaf** [reply]
I don't know if I can tell more. I know BTCPay has the option of multiple people using the same instance for onchain transactions because the instance only has to know an xpub or something like that, right?

But that doesn't work with Lightning. Only the instance owner uses the Lightning node because it has keys and other stuff in there.

But if there is the concept of accounts already maybe that same Lightning node could be shared custodially across multiple accounts? This is sounding like going over the BTCPay scope, but who knows?

**Reply by Zaxounette** [reply]
@fiatjaf A shared LN node is possible with LNbits.
We're having trouble integrating LNbits into BTCPayServer as is, but other solutions are in the works.

**Reply by fiatjaf** [reply]
Why are you having trouble? @Kukks opened an issue about this many decades ago on the LNbits repo asking for a specific feature. That feature was implemented but he never said anything again. Does that mean you're working on some integration right now? I didn't know that.

**Reply by Zaxounette** [reply]
We're working on a solution to make the LN node shared yes.
No ETA though.
